app/views/shared/file_preview/_content.html.erb

```diff
@@ -1,5 +1,5 @@
 <div class="modal-header">
-  <span class="file-name"><%= asset.file_name %></span>
+  <span class="file-name"><%= asset.file&.metadata[:name] || asset.file_name %></span>
```

Comment:
this will result in an error if there is no .metadata present (as it will try to access `[:name]` on `nil`)